build/configs/common_download.sh

```diff
@@ -43,7 +43,15 @@ if [ -z ${DEFAULT_PORT} ]; then
 			DEFAULT_PORT=$device
 		fi
 	done
+	if [ -n ${DEFAULT_PORT} ]; then
```

Comment:
I think its better to have the following in boart_metadata.txt,

DEFAULT_PORT1 and DEFAULT_PORT2 (will be present if only required, currently its only for rtl8730e).

Then, we can write the logic as follows,

```
if (DEFAULT_PORT1 is available) {
    PORT = DEFAULT_PORT1;
} else if (DEFAULT_PORT2 defined) {
    if (DEFAULT_PORT2 is available) {
        PORT = DEFAULT_PORT2;
    else {
         exit("No available port");
    }
} else {
   exit("No available port");
}
``` 

In the current way, even for some boards like imxrt which use ttyACM0, we still attempt to download to the ttyUSB port if its present.